src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const closeTodo = () => dispatch(currentTodoSlice.actions.clearTodo(null));
+
+  useEffect(() => {
+    setLoading(true);
+    getUser(currentTodo.userId)
+      .then(data => {
+        setUser(data);
+      })
+      .catch(() => {})
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
```

Comment:
<img width="947" alt="image" src="https://github.com/mate-academy/react_redux-list-of-todos/assets/64750958/05ab44c6-069a-4ae6-b4f4-d6800a9c3bea">